sdk/python/kubeflow/training/api/training_client.py

```diff
@@ -770,9 +776,12 @@ def get_job_logs(
         replica_index: Optional[int] = None,
         follow: bool = False,
         timeout: int = constants.DEFAULT_TIMEOUT,
-    ) -> Dict[str, str]:
-        """Print the training logs for the Job. By default it returns logs from
-        the `master` pod.
+    ) -> Optional[Dict[str, str]]:
+        """Get the logs for every Training Job pod. By default it returns logs from
+        the `master` pod. Logs are returned in this format: { "pod-name": "Log data" }.
+
+        If follow = True, this function prints logs to StdOut and returns None.
```

Comment:
> What if a user wants to follow and interact with the response once the job completes?

The pods created will be garbage collected based on the job's some runPolicy such as cleanup policy.
Additionally, we cannot guarantee to get logs from completed pods since the pods are removed for some reasons such as node disruption.

Reply:
I agree, let's return logs if `follow=True` as well.
Despite on the runPolicy if would be useful for user to get the final results after running `get_job_logs`.
I made that change @tenzen-y @droctothorpe.